index.js

```diff
@@ -4,54 +4,46 @@ var fs = require('graceful-fs');
 var stripBom = require('strip-bom');
 var stripBomStream = require('strip-bom-stream');
 var File = require('vinyl');
+var pify = require('pify');
+var Promise = require('pinkie-promise');
 
-exports.read = function (pth, opts, cb) {
+exports.read = function (pth, opts) {
 	opts = opts || {};
 
-	if (typeof opts === 'function') {
-		cb = opts;
-		opts = {};
-	}
-
 	var cwd = opts.cwd || process.cwd();
 	var base = opts.base || cwd;
 
 	pth = path.resolve(cwd, pth);
 
-	fs.stat(pth, function (err, stat) {
-		if (err) {
-			cb(err);
-			return;
-		}
-
-		var file = new File({
-			cwd: cwd,
-			base: base,
-			path: pth,
-			stat: stat
-		});
+	return pify(fs.stat.bind(fs), Promise)(pth)
+		.then(function (stat) {
+			var file = new File({
+				cwd: cwd,
+				base: base,
+				path: pth,
+				stat: stat
+			});
+
+			if (opts.read === false) {
+				return [file];
+			}
 
-		if (opts.read === false) {
-			cb(null, file);
-			return;
-		}
+			if (opts.buffer === false) {
+				file.contents = fs.createReadStream(pth).pipe(stripBomStream());
+				return [file];
+			}
 
-		if (opts.buffer === false) {
-			file.contents = fs.createReadStream(pth).pipe(stripBomStream());
-			cb(null, file);
-			return;
-		}
+			return Promise.all([file, pify(fs.readFile.bind(fs), Promise)(pth)]);
```

Comment:
That looks more cleaner indeed!

Didn't want to pify the entire module just for two methods.